tiledb/ml/models/base.py

```diff
@@ -6,16 +6,42 @@
 import json
 import platform
 
+from typing import Optional, Union
 from enum import Enum, unique
 
+from torch.nn import Module
+from tensorflow.python.keras.models import Model
+from sklearn.base import BaseEstimator
+
+
+@unique
+class ModelFileProperties(Enum):
+    """
+    Enum Class that contains all model array file properties.
+    """
+
+    ML_FRAMEWORK = "ML_FRAMEWORK"
+    ML_FRAMEWORK_VERSION = "ML_FRAMEWORK_VERSION"
+    STAGE = "STAGE"
+    PYTHON_VERSION = "PYTHON_VERSION"
+    PREVIEW = "PREVIEW"
+
 
 class TileDBModel(abc.ABC):
     """
     This is the base class for all TileDB model storage functionalities, i.e,
     store machine learning models (Tensorflow, PyTorch, etc) as TileDB arrays.
     """
 
-    def __init__(self, uri: str, namespace: str = None, ctx: tiledb.Ctx = None):
+    def __init__(
+        self,
+        uri: str,
+        namespace: str = None,
+        ctx: tiledb.Ctx = None,
+        model: Optional[Union[Module, Model, BaseEstimator]] = None,
+        framework: str = None,
+        framework_version: str = None,
```

Comment:
These two should be required to not be None (or annotated as Optional otherwise).